src/compute.ts

```diff
@@ -13,50 +13,79 @@
 // limitations under the License.
 
 import * as aws from "@pulumi/aws";
+
 import { EnforcementLevel, ResourceValidationPolicy, validateTypedResource } from "@pulumi/policy";
 
-export const compute: ResourceValidationPolicy[] = [
-    ec2InstanceDetailedMonitoringEnabled("mandatory"),
-    ec2InstanceNoPublicIP("mandatory"),
-    ec2VolumeInUseCheck("mandatory", true),
-    elbAccessLoggingEnabled("mandatory"),
-    encryptedVolumes("mandatory"),
-];
+import { registerPolicy } from "./awsGuard";
+import { defaultEnforcementLevel } from "./enforcementLevel";
+import { PolicyArgs } from "./policyArgs";
+import { getValueOrDefault } from "./util";
+
+// Mixin additional properties onto AwsGuardArgs.
+declare module "./awsGuard" {
+    interface AwsGuardArgs {
+        ec2InstanceDetailedMonitoringEnabled?: EnforcementLevel;
+        ec2InstanceNoPublicIP?: EnforcementLevel;
+        ec2VolumeInUseCheck?: EnforcementLevel | Ec2VolumeInUseCheckArgs;
+        elbAccessLoggingEnabled?: EnforcementLevel;
+        encryptedVolumes?: EnforcementLevel | EncryptedVolumesArgs;
```

Comment:
I like the conciseness of how you are declaring the `AwsGuardArgs` args here using mix-ins. But it has the problem of being really confusing as more and more policies are added to the policy pack. A value like `encryptedVolumes` not not be unique across all args to all policies.... and the alternative of prefixing the arg with the policy name is kinda lame too.

All this to say: I like the pattern, but I am not sure if it is going to work out for AWS Guard given the large number of rules we plan on adding.

Every alternative I can think of is strictly worse, so let's keep this as-is. Just want calling it out in case this does end up causing a lot of friction, that we should be willing to switch to one of the alternative interfaces you considered if it becomes a big problem. (But it could be I'm being paranoid, and so let's try it this way as it makes policies much easier to author.)